sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DistinctDocumentQueryExecutionContext.java

```diff
@@ -20,7 +19,9 @@
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.BiFunction;
 
-public class DistinctDocumentQueryExecutionContext<T extends Resource> implements IDocumentQueryExecutionComponent<T> {
+public class DistinctDocumentQueryExecutionContext<T>
```

Comment:
By theory, the T here can not be any type right? It has to be Document as well
Since for example aggregrateExecutionContext can use this one as the base component.

Reply:
Discussed offline, so there will be two types of pipelines.

If the queryInfo contains aggregates, orderly etc, they Document type will be force through
else, T will be used across